### PR TITLE
Wildcard V2: <Icon /> Codemod

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description

- Refactor `Icon` component implementation to remove `svg` ReactNode prop, use `as` to set svg component

## Refs
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/27669)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27669)

### CODE REVIEW

File: client/wildcard/src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -18,17 +19,14 @@ interface IconProps {
...
+    ({ children, className, size, as: Component = 'div', ...attributes }, reference) => (
+        <Component
+            className={classNames(styles.iconInline, size === 'md' && styles.iconInlineMd, className)}
+            ref={reference}
+            {...attributes}
+        >
+            {children}
+        </Component>
+    )
+) as ForwardReferenceComponent<'div', IconProps>
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.tsx:32]: Does it make sense to use `svg` here as a default?

File: client/wildcard/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -18,17 +19,14 @@ interface IconProps {
...
-}) => (
-    <Component
-        className={classNames(styles.iconInline, size === 'md' && styles.iconInlineMd, className)}
-        {...attributes}
-    >
-        {svg}
-    </Component>
-)
+export const Icon = React.forwardRef(
+    ({ children, className, size, as: Component = 'div', ...attributes }, reference) => (
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.tsx:23]: And here too

File: client/wildcard/src/components/Icon/Icon.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import classNames from 'classnames'
 import React from 'react'
 
+import { ForwardReferenceComponent } from '../..'
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.tsx:4]:
```suggestion
import { ForwardReferenceComponent } from '../../types'
```

**Review by valerybugakov — CHANGES_REQUESTED** [review]

File: client/wildcard/src/components/Icon/Icon.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/wildcard/src/components/Icon/Icon.tsx]: Thanks @valerybugakov, we got some problem with typing when using `as ForwardReferenceComponent<...>` so have to add types for `forwardRef` as alternative. Could you take a look

**Pull request merged**